src/main/java/com/exactpro/th2/service/AbstractGrpcService.java

```diff
@@ -111,14 +115,18 @@ public void onError(Throwable t) {
             int attempt = currentAttempt.getAndIncrement();
             if (attempt < retryPolicy.getMaxAttempts() && t instanceof StatusRuntimeException) {
 
-                logger.warn("Can not send GRPC async request. Retrying. Current attempt = {}", currentAttempt.get() + 1, t);
+                if (((StatusRuntimeException)t).getStatus() == Status.UNKNOWN) { // Server side thrown an exception
```

Comment:
Are we sure `t` is alwayse `StatusRuntimeException`? I think we should check it.

Reply:
We checked it in the if block
if (attempt < retryPolicy.getMaxAttempts() && **t instanceof StatusRuntimeException**) {